Sources/Plasma/FeatureLib/pfPython/pyImageLibModGlue.cpp

```diff
@@ -0,0 +1,190 @@
+non-source form of such a combination shall include the source code for
+the parts of OpenSSL and IJG JPEG Library used as well as that of the covered
+work.
+
+You can contact Cyan Worlds, Inc. by email legal@cyan.com
+ or by snail mail at:
+      Cyan Worlds, Inc.
+      14617 N Newport Hwy
+      Mead, WA   99021
+
+*==LICENSE==*/
+
+
+#include <Python.h>
+#include "pyKey.h"
+
+#include "pyImageLibMod.h"
+#include "pyImage.h"
+#include "plGImage/plMipmap.h"
+
+// glue functions
+PYTHON_CLASS_DEFINITION(ptImageLibMod, pyImageLibMod);
+
+PYTHON_DEFAULT_NEW_DEFINITION(ptImageLibMod, pyImageLibMod)
+PYTHON_DEFAULT_DEALLOC_DEFINITION(ptImageLibMod)
+
+PYTHON_INIT_DEFINITION(ptImageLibMod, args, keywords)
+{
+    PyObject* keyObj = nullptr;
+    if (!PyArg_ParseTuple(args, "O", &keyObj))
+    {
+        PyErr_SetString(PyExc_TypeError, "__init__ expects a ptKey");
+        PYTHON_RETURN_INIT_ERROR;
+    }
+    if (!pyKey::Check(keyObj))
+    {
+        PyErr_SetString(PyExc_TypeError, "__init__ expects a ptKey");
+        PYTHON_RETURN_INIT_ERROR;
+    }
+    pyKey* key = pyKey::ConvertFrom(keyObj);
+    self->fThis->setKey(*key);
+    PYTHON_RETURN_INIT_OK;
+}
+
+PYTHON_RICH_COMPARE_DEFINITION(ptImageLibMod, obj1, obj2, compareType)
+{
+    if ((obj1 == Py_None) || (obj2 == Py_None) || !pyImageLibMod::Check(obj1) || !pyImageLibMod::Check(obj2))
+    {
+        // if they aren't the same type, they don't match, obviously (we also never equal none)
+        if (compareType == Py_EQ)
+            PYTHON_RCOMPARE_FALSE;
+        else if (compareType == Py_NE)
+            PYTHON_RCOMPARE_TRUE;
+        else
+        {
+            PyErr_SetString(PyExc_NotImplementedError, "invalid comparison for a ptImageLibMod object");
+            PYTHON_RCOMPARE_ERROR;
+        }
+    }
+    pyImageLibMod* ilm1 = pyImageLibMod::ConvertFrom(obj1);
+    pyImageLibMod* ilm2 = pyImageLibMod::ConvertFrom(obj2);
+    if (compareType == Py_EQ)
+    {
+        if ((*ilm1) == (*ilm2))
+            PYTHON_RCOMPARE_TRUE;
+        PYTHON_RCOMPARE_FALSE;
+    }
+    else if (compareType == Py_NE)
+    {
+        if ((*ilm1) != (*ilm2))
+            PYTHON_RCOMPARE_TRUE;
+        PYTHON_RCOMPARE_FALSE;
+    }
+    PyErr_SetString(PyExc_NotImplementedError, "invalid comparison for a ptImage object");
+    PYTHON_RCOMPARE_ERROR;
+}
+
+PYTHON_METHOD_DEFINITION(ptImageLibMod, getImage, args)
+{
+    ST::string name;
+    if (!PyArg_ParseTuple(args, "O&", PyUnicode_STStringConverter, &name)) {
+        PyErr_SetString(PyExc_TypeError, "getImage expects an image name");
+        PYTHON_RETURN_ERROR;
+    }
+
+    plMipmap* image = plMipmap::ConvertNoRef(self->fThis->GetImage(name));
+    if (image)
+        return pyImage::New(image);
+
+    PYTHON_RETURN_NONE;
+}
+
+PYTHON_METHOD_DEFINITION_NOARGS(ptImageLibMod, getNames)
+{
+    std::vector<PyObject*> nameList = self->fThis->GetImageNames();
+    PyObject* retVal = PyList_New(nameList.size());
```

Comment:
This should probably return a tuple. Returning a list indicates that what you're returning is mutable.

Reply:
Good catch.  That's what I *intended* to do, but did not.